python/src/aac/io/parser/_cache.py

```diff
@@ -41,9 +41,9 @@ class YamlLFUCache:
     # The internal cache is using a dict with sorting O(n log n) rather than a more complex linked list which would be capable of linear time
     cache: dict[str, CacheEntry] = attrib(default=Factory(dict), validator=validators.instance_of(dict))
 
-    def parse_string(self, string: str) -> list[dict]:
+    def parse_string(self, string: str, source: str = STRING_YAML_SOURCE) -> list[dict]:
```

Comment:
I had to add the ability to override string sources for testing and the revised parser functionality.